src/lib/services-manager/dialogs/services_manager.rb

```diff
@@ -303,27 +302,33 @@ def abort_handler
 
       # Handler for next event (accept button is used)
       #
-      # @note It finishes the dialog if all changes were correctly applied.
+      # @note A confirmation popup is shown and it finishes the dialog if all
+      #   changes were correctly applied.
       def next_handler
+        return unless confirm_changes?
+
         self.success = save
 
         if !success && continue_editing?
           self.finish = false
-          refresh_services
+          refresh
         else
           self.finish = true
         end
       end
 
       # Handler for apply event (apply button is used)
       #
-      # @note It does not finish the dialog when all changes were correctly applied.
+      # @note A confirmation popup is shown and it does not finish the dialog when
```

Comment:
NP: "could be shown", since the confirmation popup will be displayed only if there are changes. Isn't it?

Reply:
In this case the dialog will be always shown because the Apply button is enabled only if there is any change. Anyway, as you commented, the comment needs to be adapted in for `#next_handler` due to "Ok" button is always available (even if there is no change at all).